### PR TITLE
fix: improve message display for typed sign messages

### PR DESCRIPTION
**Pull request by jpuri**
## **Description**
Improve message display for typed sign messages.

## **Related issues**
NA

## **Manual testing steps**

1. Go to test dapp
2. Submit typed sign
3. Message should be well formatted

## **Screenshots/Recordings**
Screenshot 2024-05-30 at 6 54 14 PM

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/util/string/index.ts [hunk 1]
```diff
@@ -16,3 +16,123 @@ export default (str: any): any => {
...
+      types,
+    );
+  });
+  return { value: sanitizedStruct, type: primaryType };
+};
+
+export const parseTypedSignDataMessage = (dataToParse: string) => {
+  const { message, primaryType, types } = JSON.parse(dataToParse);
+  return sanitizeMessage(message, primaryType, types);
+};
```
Comment (jpuri) [inline app/util/string/index.ts:138]: Code above is taken from extension utility: https://github.com/MetaMask/metamask-extension/blob/13bd82914d1ac2c584fb799e6eee9b90fe889fe2/ui/helpers/utils/util.js#L455

**Comment by sleepytanya** [comment]
@jpuri Looks good?

https://github.com/MetaMask/metamask-mobile/assets/104780023/b88a9b43-da42-410e-9ba2-85060d1d7bff

**Review by vinistevam — DISMISSED** [review]

**Review by matthewwalsh0 — DISMISSED** [review]

File: app/util/string/index.text.ts [hunk 2]
```diff
@@ -1,12 +1,72 @@
-import sanitizeString from '.';
```
Comment (tommasini) [inline app/util/string/index.text.ts:1]: I think this file name is meant to be `.test.ts` instead of `.text.ts` right?

**Review by vinistevam — DISMISSED** [review]

**Review by matthewwalsh0 — DISMISSED** [review]

File: app/util/string/index.ts [hunk 3]
```diff
@@ -16,3 +16,123 @@ export default (str: any): any => {
...
+const stripArrayType = (potentialArrayType: string) =>
+  potentialArrayType.replace(/\[[[0-9]*\]*/gu, '');
+
+const stripOneLayerofNesting = (potentialArrayType: string) =>
+  potentialArrayType.replace(/\[(\d*)\]/u, '');
+
+const isArrayType = (potentialArrayType: string) =>
+  potentialArrayType.match(/\[[[0-9]*\]*/u) !== null;
+
+const isSolidityType = (type: any) => SOLIDITY_TYPES.includes(type);
```
Comment (tommasini) [inline app/util/string/index.ts:83]: Can we avoid anys? Sorry missed this on the first review!

**Review by matthewwalsh0 — APPROVED** [review]

**Review by vinistevam — APPROVED** [review]

**Review by tommasini — APPROVED** [review]
Awesome! Thanks! LGTM!

**Pull request merged**